agent/engine/task_manager_test.go

```diff
@@ -15,19 +15,382 @@
+		// when desired is Running
+		{api.ContainerPulled, api.ContainerStatusNone, api.ContainerStatusNone, false, false},
+		// PULLED -> STOPPED transition will result in STOPPED and is allowed, but not
+		// actionable, when desired is STOPPED
+		{api.ContainerPulled, api.ContainerStopped, api.ContainerStopped, false, true},
+		// CREATED -> RUNNING transition is allowed and actionable, when desired is Running
+		// The exptected next status is Running
+		{api.ContainerCreated, api.ContainerRunning, api.ContainerRunning, true, true},
+		// CREATED -> CREATED transition is not allowed and not actionable,
+		// when desired is Running
+		{api.ContainerCreated, api.ContainerCreated, api.ContainerStatusNone, false, false},
+		// CREATED -> NONE transition is not allowed and not actionable,
+		// when desired is Running
+		{api.ContainerCreated, api.ContainerStatusNone, api.ContainerStatusNone, false, false},
+		// CREATED -> PULLED transition is not allowed and not actionable,
+		// when desired is Running
+		{api.ContainerCreated, api.ContainerPulled, api.ContainerStatusNone, false, false},
+		// CREATED -> STOPPED transition will result in STOPPED and is allowed, but not
+		// actionable, when desired is STOPPED
+		{api.ContainerCreated, api.ContainerStopped, api.ContainerStopped, false, true},
+		// RUNNING -> STOPPED transition is allowed and actionable, when desired is Running
+		// The exptected next status is STOPPED
+		{api.ContainerRunning, api.ContainerStopped, api.ContainerStopped, true, true},
+		// RUNNING -> RUNNING transition is not allowed and not actionable,
+		// when desired is Running
+		{api.ContainerRunning, api.ContainerRunning, api.ContainerStatusNone, false, false},
+		// RUNNING -> NONE transition is not allowed and not actionable,
+		// when desired is Running
+		{api.ContainerRunning, api.ContainerStatusNone, api.ContainerStatusNone, false, false},
+		// RUNNING -> PULLED transition is not allowed and not actionable,
+		// when desired is Running
+		{api.ContainerRunning, api.ContainerPulled, api.ContainerStatusNone, false, false},
+		// RUNNING -> CREATED transition is not allowed and not actionable,
+		// when desired is Running
+		{api.ContainerRunning, api.ContainerCreated, api.ContainerStatusNone, false, false},
+	}
+
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("%s to %s Transition",
+			tc.containerCurrentStatus.String(), tc.containerDesiredStatus.String()), func(t *testing.T) {
+			container := &api.Container{
+				DesiredStatusUnsafe: tc.containerDesiredStatus,
+				KnownStatusUnsafe:   tc.containerCurrentStatus,
+			}
+			task := &managedTask{
+				Task: &api.Task{
+					Containers: []*api.Container{
+						container,
+					},
+					DesiredStatusUnsafe: api.TaskRunning,
+				},
+			}
+			nextStatus, actionRequired, possible := task.containerNextState(container)
+			assert.Equal(t, tc.expectedContainerStatus, nextStatus,
+				"Expected next state [%s] != Retrieved next state [%s]",
```

Comment:
Do you need the format string here?  `assert.Equal` should print out the diff if there is one.